src/bgw/job.c

```diff
@@ -1333,3 +1380,33 @@ ts_bgw_job_insert_relation(Name application_name, Interval *schedule_interval,
 	table_close(rel, NoLock);
 	return values[AttrNumberGetAttrOffset(Anum_bgw_job_id)];
 }
+
+/*
+ * This function ensures the schedule interval is acceptable in the case
+ * of fixed job schedules. Intervals that mix months with day and time
+ * components are not acceptable since internally we use time_bucket and
+ * cannot bucket by such an interval.
+ */
+void
+ts_bgw_job_validate_schedule_interval(Interval *schedule_interval)
+{
+	bool has_month, has_day, has_time;
+	has_month = schedule_interval->month;
+	has_day = schedule_interval->day;
+	has_time = schedule_interval->time;
+
+	if (has_month && (has_day || has_time))
+		ereport(ERROR,
+				errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				errmsg("month intervals cannot have day or time component"),
+				errdetail("Fixed schedule jobs do not support such schedule intervals."),
+				errhint("Express the interval in terms of days or time instead."));
+}
+
+void
+ts_bgw_job_validate_timezone(Datum timezone)
```

Comment:
```suggestion
char *
ts_bgw_job_timezone_to_valid_cstring(Datum timezone)
```

Since you always use this function the same way:
```
ts_bgw_job_validate_timezone(Datum timezone);
valid_timezone = text_to_cstring(timezone);
```

You could make it return the C-string itself.